multicluster/apis/multicluster/v1alpha1/gateway_types.go

```diff
@@ -25,6 +25,12 @@ type GatewayInfo struct {
 	GatewayIP string `json:"gatewayIP,omitempty"`
 }
 
+// WireGuardInfo includes information of a WireGuard tunnel.
+type WireGuardInfo struct {
+	PublicKey          string `json:"publicKey,omitempty"`
+	WireGuardGatewayIP string `json:"wireGuardGatewayIP,omitempty"`
```

Comment:
Doesn't we deduct the IP from PodCIDR? I feel no need to report the IP to avoid exposing this concept to users, anyway it's only used internally.

Reply:
The WireGuard client needs to add each peer's IP address to `allowed-ips`. So we use the first IP of the PodCIDRs as the WireGuard interface IP, we still need to export it to peer. Or the packet will be dropped.

Reply:
I mean the peer already knows the PodCIDR and deduct the IP from it? no need to expose a redundant information which is used internally.

Reply:
The PodCIDR is an optional field in ClusterInfoImport, it's only exported when Pod-to-Pod connectivity is enabled. So the peer cannot calculate the WireGuard interface' IP address in general.

Reply:
Still trying to understand it. If PodCIDR is not provided, how does we know which WireGuard GatewayIP should be used? If it's provided, why we don't just fill it in the Gateway API but add another field?

Reply:
I used nodeConfig.IPv4.IP as the WireGuard interface IP address, so we just need to export the WireGuard interface IP address in `ClusterInfoImport.WireGuard.WireGuardGatewayIP`. The reason why we don't just fill it in the GatewayIP field is that we need two IP addresses, one is the "WireGuard endpoint",  which is the real remote Gateway IP. Another is the IP address allocated to the WireGuard interface, which is "allowed-ip".

Reply:
Change to Service CIDR's first IP now.